cavern/src/main/java/org/opencadc/cavern/nodes/AclCommandExecutor.java

```diff
@@ -0,0 +1,191 @@
+    private static final String FILE_RO = "r--";
+    private static final String FILE_RW = "rw-";
+    private static final String DIR_RO = "r-x";
+    private static final String DIR_RW = "rwx";
+    
+    
+    
+    private final Path path;
+    UserPrincipalLookupService users;
+    
+    public AclCommandExecutor(Path path, UserPrincipalLookupService users) { 
+        this.path = path;
+        this.users = users;
+    }
+    
+    public void clearACL()  throws IOException {
+        
+        String[] cmd = new String[] {
+            SETACL, "-b", toAbsolutePath(path)
+        };
+        BuilderOutputGrabber grabber = new BuilderOutputGrabber();
+        grabber.captureOutput(cmd);
+        if (grabber.getExitValue() != 0) {
+            throw new IOException("failed to clear ACLs on " + path + ": "
+                + grabber.getErrorOutput(true));
+        }
+    }
+    
+    public void setReadOnlyACL(GroupPrincipal group, boolean isDir) throws IOException {
+        String perm = FILE_RO;
+        if (isDir) {
+            perm = DIR_RO;
+        }
+        setACL(group, perm);
+    }
+    
+    public void setReadWriteACL(GroupPrincipal group, boolean isDir) throws IOException {
+        String perm = FILE_RW;
+        if (isDir) {
+            perm = DIR_RW;
+        }
+        setACL(group, perm);
+    }
+    
+    private void setACL(GroupPrincipal group, String perm) throws IOException {
+        String[] cmd = new String[] {
+            SETACL, "-m", "group:" + group.getName() + ":" + perm, toAbsolutePath(path)
+        };
+        BuilderOutputGrabber grabber = new BuilderOutputGrabber();
+        grabber.captureOutput(cmd);
+        if (grabber.getExitValue() != 0) {
+            throw new IOException("failed to set read-only ACL on " + path + ": "
+                + grabber.getErrorOutput(true));
+        }
+    }
+    
+    public GroupPrincipal getReadOnlyACL(boolean isDir) throws IOException {
+        String perm = FILE_RO;
+        if (isDir) {
+            perm = DIR_RO;
+        }
+        return getACL(perm);
+    }
+    
+    public GroupPrincipal getReadWriteACL(boolean isDir) throws IOException {
+        String perm = FILE_RW;
+        if (isDir) {
+            perm = DIR_RW;
+        }
+        return getACL(perm);
+    }
+    
+    private GroupPrincipal getACL(String perm) throws IOException {
+        String[] cmd = new String[] {
+            GETACL, toAbsolutePath(path)
+        };
+        BuilderOutputGrabber grabber = new BuilderOutputGrabber();
+        grabber.captureOutput(cmd);
+        if (grabber.getExitValue() != 0) {
+            throw new IOException("failed to set read-only ACL on " + path + ": "
+                + grabber.getErrorOutput(true));
+        }
+        String out = grabber.getOutput(true);
+        String[] lines = out.split("[\n]");
+        for (String s : lines) {
+            if (!s.startsWith("#")) {
+                String[] tokens = s.split(":");
```

Comment:
Think this acl output is fairly standard?

Reply:
I sure hope so :-) All linux systems will use the same command line tools for this. I guess some options would make parsing more robust: -c (omit header) and -s (no output if only base entries present). Something to do in in the refactor/cleanup phase.

Reply:
waiting for an hour-long job to run so I will improve this a little...